Assets/MixedRealityToolkit/_Core/Definitions/InteractionDefinition.cs

```diff
@@ -23,5 +27,93 @@ public struct InteractionDefinition
         /// The primary action of the button as defined by the controller SDK.
         /// </summary>
         public InputType InputType { get; set; }
+
+        private object rawDatum { get; set; }
+        private bool boolDatum { get; set; }
+        private float floatDatum { get; set; }
+        private Vector2 vector2Datum { get; set; }
+        private Vector3 positionDatum { get; set; }
+        private Quaternion rotationDatum { get; set; }
+
+        public bool Changed { get; set; }
+
+        public object GetValue()
+        {
+            switch (AxisType)
+            {
+                case AxisType.Digital:
+                    return boolDatum;
+                case AxisType.SingleAxis:
+                    return floatDatum;
+                case AxisType.DualAxis:
+                    return vector2Datum;
+                case AxisType.ThreeDoF:
+                    return rotationDatum;
+                case AxisType.SixDoF:
+                    return new Tuple<Vector3, Quaternion>(positionDatum, rotationDatum);
+                case AxisType.Raw:
+                case AxisType.None:
+                default:
+                    return rawDatum;
+            }
+        }
+
+        public void SetValue(object value)
+        {
+            switch (AxisType)
+            {
+                case AxisType.Raw:
+                    Changed = value == rawDatum;
+                    rawDatum = value;
+                    break;
+                case AxisType.Digital:
+                    bool newBoolValue;
+                    Assert.IsTrue(bool.TryParse(value.ToString(), out newBoolValue));
```

Comment:
Should probably use the `UnityEngine.Debug.Assert` if this isn't a NUnit test class

Reply:
Used the same method as used in other parts of the toolkit. Will review and update.

I don't particularly like this approach, but it will do until "I can find something better"

Reply:
Werent we going to have Assert wrappers? Then we could have Unity and runtime asserts.... Please.

Reply:
I _really_ *really* disagree with wrapping asserts and debug logs.

Reply:
In either case, I was just trying to point out that Simon used the wrong namespace for the asserts.

Reply:
Will review.  ANd I know you are against Wrappers @StephenHodgson but it's much better to control that from a Single class and use our own implementation, rather than spat Unity stuff everywhere as it leaves us open to issues if Unity introduce a breaking change.  I prefer to be in control, not unity.

Reply:
Well unfortunately that impacts others who use the toolkit.  I don't like when dependencies change the way logs and assertions are handled.  For example, we have tools and things we use that are directly impacted by a change like this.

Reply:
We're using Unity, do it the Unity way. Accept the breaking changes when they come. Not _if_ they come.

Reply:
On the namespace thing.  It IS using the UnityEngine.Debug.Assert. in the same way you do in the Input manager :D

P.S. Wrappers would make this MORE clear :D